applications/CoSimulationApplication/python_scripts/coupled_solvers/gauss_seidel_strong.py

```diff
@@ -102,15 +102,16 @@ def SolveSolutionStep(self):
                 if self.echo_level > 0:
                     cs_tools.cs_print_info(self._ClassName(), colors.green("### CONVERGENCE WAS ACHIEVED ###"))
                 return True
-            else:
-                # TODO I think this should not be done in the last iterations if the solution does not converge in this timestep
```

Comment:
this PR basically implements what I was thinking when I initially wrote it but didn't have the time to further investigate
So far it did not really play a role since (at least for me) the simulations converged